File: app/src/main/java/com/waveneuro/domain/usecase/patient/GetOrganizationsUseCase.java
package com.waveneuro.domain.usecase.patient;

import com.asif.abase.domain.base.ObservableUseCase;
import com.waveneuro.data.DataManager;
import com.waveneuro.data.model.response.patient.PatientListResponse;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.rxjava3.core.Observable;

public class GetOrganizationsUseCase extends ObservableUseCase<List<PatientListResponse.Patient.Organization>> {

    private final DataManager dataManager;

    @Inject
    public GetOrganizationsUseCase(DataManager dataManager) {
        this.dataManager = dataManager;
    }

    @Override
    public Observable<List<PatientListResponse.Patient.Organization>> buildUseCaseSingle() {
        return dataManager.organizations();
    }


}

File: app/src/main/java/com/waveneuro/domain/usecase/client/GetOrganizationsUseCase.kt
package com.waveneuro.domain.usecase.client

import com.asif.abase.domain.base.ObservableUseCase
import com.waveneuro.data.DataManager
import com.waveneuro.data.model.response.organization.OrganizationResponse
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class GetOrganizationsUseCase @Inject constructor(
    private val dataManager: DataManager
) : ObservableUseCase<List<OrganizationResponse>>() {

    override fun buildUseCaseSingle(): Observable<List<OrganizationResponse>> {
        return dataManager.organizations()
    }

}

